Added AIKONF4

DieHertz opened the pull request:
The target I've been helping AIKON with.

* STM32F405
* MPU6000, trying to convince AIKON to go with ICM-20602 or ICM-20608-G with Tantalum caps for filtering
* AT7456E OSD
* 4-6 motors
* Timer assignment most suitable for burst dshot and with possibility to run softserial on about any spare UART or MOTOR pin
* 11-pin connector for AIKON 4in1 32-bit ESC to provide voltage, current and telemetry readings
* Dedicated camera control pin
* PPM shared with LEDSTRIP
* 16 MiB SPI FLASH, I'm working on convincing AIKON to add 32 or 64 MiB in final version

mikeller commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define USBD_PRODUCT_STRING "AIKONF4"
+
+#ifdef OPBL
+#define USBD_SERIALNUMBER_STRING "0x8020000" // Remove this at the next major release (?)
```
Are there plans to release an OPBL version for this? If not this is not needed.

DieHertz replied:
No plans, will remove it

mikeller commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define BEEPER                    PB5
+#define BEEPER_INVERTED
+
+#define USE_DSHOT_DMAR
```
Not needed.
I think you want #define ENABLE_DSHOT_DMAR true instead.

DieHertz replied:
Thanks
Indeed, it was a bit outdated, from the days before `dshot_burst` CLI setting

mikeller commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define SERIALRX_UART           SERIAL_PORT_USART1
+
+#define DEFAULT_RX_FEATURE          FEATURE_RX_SERIAL
+#define DEFAULT_FEATURES            ( FEATURE_OSD | FEATURE_SOFTSERIAL  )
```
Spacing.

hydra replied:
two spaces near the end of the line

mikeller commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define TARGET_IO_PORTD BIT(2)
+
+#define USABLE_TIMER_CHANNEL_COUNT 16
+#define USED_TIMERS  (  TIM_N(1) | TIM_N(2) | TIM_N(3) | TIM_N(4) | TIM_N(5) | TIM_N(8) | TIM_N(9))
```
Spacing.

DieHertz replied:
You mean `BIT(2)` not being indented to `16` level?

wind0r replied:
Spacing aka Whitespace :P 
(  TIM_N(1)

mikeller replied:
Yep. Double space before, no space after. But this is just nitpicking.

hydra commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+
+#define USE_BARO
+#define USE_BARO_BMP280
+// @note AIKON uses PB7 as CS for MBP280, but BF only uses CS for SPI
```
typo in sensor name in comment

hydra commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    SPI3
+#define MAX7456_SPI_CS_PIN      SPI3_NSS_PIN
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD) // 10MHz
```
probably best to remove the comment about 10mhz

hydra commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+#define SERIAL_PORT_COUNT       7 // VCP, USART1, USART3,USART4, USART6, SOFT_SERIAL1, SOFT_SERIAL2
+
+#define USE_ESCSERIAL
+//@todo redo
```
todo left behind

hydra commented on src/main/target/AIKONF4/target.h:
```diff
@@ -0,0 +1,143 @@
+
+#define USE_I2C
+// @note is this necessary?
+//#define USE_I2C_PULLUP
```
i guess not needed 😄  probably you can delete the comment

DieHertz commented:
Corrected whitespace and removed stale comments. Will work on a wiki page in the coming days. Also need to grab my Unify to check that all advertised features work as intended.

mikeller commented:
@DieHertz: Is this still in development?

DieHertz commented:
Yes, I'm waiting for the final prototype.

mikeller commented:
Ah, I see. I am holding my breath for the Russian mail service to work fast (or rather I'm not). :-P

DieHertz commented:
Added a [wiki page](https://github.com/betaflight/betaflight/wiki/Aikon-F4), will perform the final checks tonight and add a few photos and maybe a wiring diagram.

mikeller commented on src/main/target/AIKONF4/target.c:
```diff
@@ -0,0 +1,53 @@
+    DEF_TIM(TIM3,  CH4,  PB1,  TIM_USE_MOTOR,               0, 0), // M6
+
+    // backdoor timers
+    DEF_TIM(TIM1,  CH2,  PA9,  TIM_USE_ANY,                 0, 0), // UART1_TX
```
Does it make sense to define backdoor timers on pins that are connected to an inverter? What can these pins be used for (except for being used by the UART controlling the inverter)?

DieHertz replied:
Indeed doesn't, removed it.

mikeller commented:
Added link to wiki.
